### PR TITLE
chore: rename app from osmo-notify to osmoX

### PR DESCRIPTION
**Pull request by LakshayaT**
**Description:**
This PR renames all occurrences of osmo-notify, Osmo-Notify and Osmo Notify to osmox, osmoX and OsmoX as required.

**Related changes:**
_Documentation:_
- Rename collection name in PR templates
- Rename occurrences in CONTRIBUTING.md, README.md (base, api and portal)
- Rename occurrences in api docs like add-new-provider.md, api-test-cases.md, block-diagram.md, development-setup.md, production-setup.md, usage-guide.md

_Codebase:_
- Update name in api logger.config.ts
- Update occurrences in api notifications.service.ts
- Update heading in app.component.html
- Update title in app.component.ts, app.component.spec.ts and index.html
- Update localStorage item names in auth.service.ts, login.component.ts and graphql.service.ts

_Others:_
- Rename postman collection to OsmoX.postman.collection.json
- Rename block diagram file to OsmoX-block-diagram.png
- Update occurrences in docker-compose.yml (api and portal)
- Update name to OsmoX in ecosystem.config.js
- Update occurrences in package.json and package-lock.json (api and portal)
- Update occurrences in angular.json
- Update repo link references to osmo-x

**Screenshots:**
image

## Summary by CodeRabbit

- **Documentation**
	- Updated project documentation and guides to reflect the rebranding from "Osmo Notify" to "OsmoX."
- **New Features**
	- Rebranded the application, including all references and component names, to "OsmoX" to align with the new branding strategy.
- **Refactor**
	- Updated internal references, configuration variables, and storage keys from "Osmo Notify" to "OsmoX" across various files and documents.
- **Style**
	- Modified the application and portal titles to "OsmoX" in user-facing interfaces.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template_api.md [hunk 1]
```diff
@@ -3,9 +3,9 @@
 ### Pre-requisites
 
 - [ ] I have gone through the Contributing guidelines for [Submitting a Pull Request (PR)](../../CONTRIBUTING.md#submitting-a-pull-request-pr) and ensured that this is not a duplicate PR.
-- [ ] I have performed preliminary testing using the [test suite](../../apps/api/osmo-notify.postman_collection.json) to ensure that any existing features are not impacted and any new features are working as expected.
+- [ ] I have performed preliminary testing using the [test suite](../../apps/api/osmoX.postman_collection.json) to ensure that any existing features are not impacted and any new features are working as expected.
```
Comment (xixas) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template_api.md:6]: You can fix the casing for this as `OsmoX.postman_collection.json`

File: apps/api/docker-compose.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 version: '3'
 services:
```
Comment (xixas) [inline apps/api/docker-compose.yml:2]: Did you check the docker running after making these change?

File: apps/api/docs/api-test-cases.md [hunk 3]
```diff
@@ -30,7 +30,7 @@ Before running the test cases, make sure you have the following prerequisites in
 
    1. Open Postman.
    2. Click on the `Import` button.
-   3. Select the `apps/api/osmo-notify.postman_collection.json` file.
+   3. Select the `apps/api/osmoX.postman_collection.json` file.
```
Comment (xixas) [inline apps/api/docs/api-test-cases.md:33]: You can fix the casing for this as `OsmoX.postman_collection.json`

File: apps/api/docs/api-test-cases.md [hunk 4]
```diff
@@ -42,7 +42,7 @@ Before running the test cases, make sure you have the following prerequisites in
    Alternatively, click the three dots beside the collection name and click `Run collection`
 
 3. Select the environment on top right.
-4. Click `Run osmo-notify` to execute the test cases.
+4. Click `Run osmoX` to execute the test cases.
```
Comment (xixas) [inline apps/api/docs/api-test-cases.md:45]: Update collection name(assuming it takes it from file name)

File: apps/api/docs/api-test-cases.md [hunk 5]
```diff
@@ -52,13 +52,13 @@ Before running the test cases, make sure you have the following prerequisites in
     1. For console output
 
        ```bash
-       newman run osmo-notify.postman_collection.json -e osmo-notify.postman_environment.json
+       newman run osmoX.postman_collection.json -e osmoX.postman_environment.json
```
Comment (xixas) [inline apps/api/docs/api-test-cases.md:55]: You can fix the casing for this as `OsmoX.postman_environment.json`

File: apps/api/docs/api-test-cases.md [hunk 6]
```diff
@@ -52,13 +52,13 @@ Before running the test cases, make sure you have the following prerequisites in
...
        ```
 
     2. For HTML report
 
        ```bash
-       newman run osmo-notify.postman_collection.json -e osmo-notify.postman_environment.json -r html
+       newman run osmoX.postman_collection.json -e osmoX.postman_environment.json -r html
```
Comment (xixas) [inline apps/api/docs/api-test-cases.md:61]: You can fix the casing for this as `OsmoX`

File: apps/api/docs/block-diagram.md [hunk 7]
```diff
@@ -2,21 +2,21 @@
...
+This document provides a concise block diagram illustrating the architecture of our software application, OsmoX. The diagram outlines the key components and their interactions to help you better understand how the application functions.
 
 ## Block Diagram Overview
 
 The block diagram represents the software application's architecture in a visual format, making it easier to grasp its structure and functionality.
 
 ## Block Diagram
 
-![Block Diagram](./assets/osmo-notify-block-diagram.png)
+![Block Diagram](./assets/osmoX-block-diagram.png)
```
Comment (xixas) [inline apps/api/docs/block-diagram.md:13]: make it  "OsmoX-block-diagram.png."

File: apps/api/docs/development-setup.md [hunk 8]
```diff
@@ -40,22 +40,22 @@ Make sure Redis and MariaDB server are up and running.
    ```env
    # Server
    SERVER_PORT=3000
-   SERVER_API_KEY=osmo-notify-test-key
+   SERVER_API_KEY=osmoX-test-key
```
Comment (xixas) [inline apps/api/docs/development-setup.md:43]: here also make it `OsmoX` as this is more of a reference to OsmoX not value like `osmox-mariadb`.

File: apps/api/docs/development-setup.md [hunk 9]
```diff
@@ -81,7 +81,7 @@ Make sure Redis and MariaDB server are up and running.
    TWILIO_WA_NUMBER=
 
    # TEMP
-   APP_NAME=osmo_notify
+   APP_NAME=osmoX
```
Comment (xixas) [inline apps/api/docs/development-setup.md:84]: Make it `OsmoX`

File: apps/api/docs/production-setup.md [hunk 10]
```diff
@@ -24,19 +24,19 @@ Make sure Redis and MariaDB server are up and running.
   ```env
   # Server
   SERVER_PORT=3000
-  SERVER_API_KEY=osmo-notify-test-key
+  SERVER_API_KEY=osmoX-test-key
```
Comment (xixas) [inline apps/api/docs/production-setup.md:27]: Update to `OsmoX`

File: apps/api/docs/production-setup.md [hunk 11]
```diff
@@ -52,21 +52,21 @@ Make sure Redis and MariaDB server are up and running.
...
-  
+
   # WhatsApp 360Dialog
   ENABLE_WA360DIALOG=true
   WA_360_DIALOG_URL=https://waba.360dialog.io/v1/messages
   WA_360_DIALOG_API_KEY=
 
   # TEMP
-  APP_NAME=osmo_notify
+  APP_NAME=osmoX
```
Comment (xixas) [inline apps/api/docs/production-setup.md:62]: Update to `OsmoX`

File: apps/api/docs/production-setup.md [hunk 12]
```diff
@@ -82,7 +82,7 @@ Make sure to replace the above example values with appropriate values as per you
   module.exports = {
     apps: [
       {
-        name: 'osmo-notify', // Name of your application
+        name: 'osmoX', // Name of your application
```
Comment (xixas) [inline apps/api/docs/production-setup.md:85]: Update to `OsmoX`

File: apps/api/docs/usage-guide.md [hunk 13]
```diff
@@ -69,8 +71,8 @@ Replace `SERVER_API_KEY_VALUE` with the actual API key value you want to include
         "text": "This is a test notification",
         "html": "<b>This is a test notification</b>"
       },
-      "createdBy": "osmo_notify",
-      "updatedBy": "osmo_notify",
+      "createdBy": "osmoX",
```
Comment (xixas) [inline apps/api/docs/usage-guide.md:74]: Update to `OsmoX`

File: apps/api/ecosystem.config.js [hunk 14]
```diff
@@ -1,7 +1,7 @@
 module.exports = {
   apps: [
     {
-      name: 'osmo-notify', // Name of your application
+      name: 'osmoX', // Name of your application
```
Comment (xixas) [inline apps/api/ecosystem.config.js:4]: Update to OsmoX

File: apps/api/src/modules/notifications/notifications.service.ts [hunk 15]
```diff
@@ -36,8 +36,8 @@ export class NotificationsService {
       throw new BadRequestException(`Channel ${notification.channelType} is not enabled`);
     }
 
-    notification.createdBy = this.configService.getOrThrow<string>('APP_NAME') || 'osmo_notify';
-    notification.updatedBy = this.configService.getOrThrow<string>('APP_NAME') || 'osmo_notify';
+    notification.createdBy = this.configService.getOrThrow<string>('APP_NAME') || 'osmoX';
```
Comment (xixas) [inline apps/api/src/modules/notifications/notifications.service.ts:39]: Update to OsmoX

File: apps/portal/README.md [hunk 16]
```diff
@@ -1,4 +1,4 @@
-# OsmoNotifyFrontend
+# OsmoXFrontend
```
Comment (xixas) [inline apps/portal/README.md:1]: Update to OsmoX Portal, update the package.json and angular.json too, see if it takes more time, if yes do as different task instead.

**Review by xixas — CHANGES_REQUESTED** [review]

File: apps/api/docker-compose.yml (hunk 2 above)
Comment (LakshayaT) [reply apps/api/docker-compose.yml:2]: Building and running the docker did not give any errors 
image

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template_api.md (hunk 1 above)
Comment (LakshayaT) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template_api.md]: Addressed in https://github.com/OsmosysSoftware/osmo-x/pull/152/commits/ffca8627cb4ae9663e16c6d1288440193d1ada1b

File: apps/api/docs/api-test-cases.md (hunk 3 above)
Comment (LakshayaT) [reply apps/api/docs/api-test-cases.md]: Addressed in https://github.com/OsmosysSoftware/osmo-x/pull/152/commits/ffca8627cb4ae9663e16c6d1288440193d1ada1b

File: apps/api/docs/api-test-cases.md (hunk 4 above)
Comment (LakshayaT) [reply apps/api/docs/api-test-cases.md]: Addressed in https://github.com/OsmosysSoftware/osmo-x/pull/152/commits/ffca8627cb4ae9663e16c6d1288440193d1ada1b

File: apps/api/docs/api-test-cases.md (hunk 5 above)
Comment (LakshayaT) [reply apps/api/docs/api-test-cases.md]: Addressed in https://github.com/OsmosysSoftware/osmo-x/pull/152/commits/ffca8627cb4ae9663e16c6d1288440193d1ada1b

File: apps/api/docs/api-test-cases.md (hunk 6 above)
Comment (LakshayaT) [reply apps/api/docs/api-test-cases.md]: Addressed in https://github.com/OsmosysSoftware/osmo-x/pull/152/commits/ffca8627cb4ae9663e16c6d1288440193d1ada1b

File: apps/api/docs/block-diagram.md (hunk 7 above)
Comment (LakshayaT) [reply apps/api/docs/block-diagram.md]: Addressed in https://github.com/OsmosysSoftware/osmo-x/pull/152/commits/ffca8627cb4ae9663e16c6d1288440193d1ada1b

File: apps/api/docs/development-setup.md (hunk 9 above)
Comment (LakshayaT) [reply apps/api/docs/development-setup.md]: Addressed in https://github.com/OsmosysSoftware/osmo-x/pull/152/commits/ffca8627cb4ae9663e16c6d1288440193d1ada1b

File: apps/api/docs/production-setup.md (hunk 10 above)
Comment (LakshayaT) [reply apps/api/docs/production-setup.md]: Addressed in https://github.com/OsmosysSoftware/osmo-x/pull/152/commits/ffca8627cb4ae9663e16c6d1288440193d1ada1b

File: apps/api/docs/production-setup.md (hunk 12 above)
Comment (LakshayaT) [reply apps/api/docs/production-setup.md]: Addressed in https://github.com/OsmosysSoftware/osmo-x/pull/152/commits/ffca8627cb4ae9663e16c6d1288440193d1ada1b

File: apps/api/docs/usage-guide.md (hunk 13 above)
Comment (LakshayaT) [reply apps/api/docs/usage-guide.md]: Addressed in https://github.com/OsmosysSoftware/osmo-x/pull/152/commits/ffca8627cb4ae9663e16c6d1288440193d1ada1b

File: apps/api/ecosystem.config.js (hunk 14 above)
Comment (LakshayaT) [reply apps/api/ecosystem.config.js]: Addressed in https://github.com/OsmosysSoftware/osmo-x/pull/152/commits/ffca8627cb4ae9663e16c6d1288440193d1ada1b

File: apps/api/src/modules/notifications/notifications.service.ts (hunk 15 above)
Comment (LakshayaT) [reply apps/api/src/modules/notifications/notifications.service.ts]: Addressed in https://github.com/OsmosysSoftware/osmo-x/pull/152/commits/ffca8627cb4ae9663e16c6d1288440193d1ada1b

File: apps/portal/README.md (hunk 16 above)
Comment (LakshayaT) [reply apps/portal/README.md]: Addressed in https://github.com/OsmosysSoftware/osmo-x/pull/152/commits/ffca8627cb4ae9663e16c6d1288440193d1ada1b

File: apps/api/docs/production-setup.md (hunk 11 above)
Comment (LakshayaT) [reply apps/api/docs/production-setup.md]: Addressed in https://github.com/OsmosysSoftware/osmo-x/pull/152/commits/ffca8627cb4ae9663e16c6d1288440193d1ada1b

File: apps/api/docs/development-setup.md (hunk 8 above)
Comment (LakshayaT) [reply apps/api/docs/development-setup.md]: Addressed in https://github.com/OsmosysSoftware/osmo-x/pull/152/commits/ffca8627cb4ae9663e16c6d1288440193d1ada1b

**Review by osm-vishnukyatannawar — APPROVED** [review]

**Pull request merged**